baseclasses/testing/pyRegTest.py

```diff
@@ -263,23 +263,25 @@ def par_add_norm(self, name, values, **kwargs):
     # Private functions
     # *****************
     def assert_allclose(self, actual, reference, name, rtol, atol, full_name=None):
+        """This is basically a wrapper on numpy.testing.assert_allclose with a generated error message"""
         if full_name is None:
             full_name = name
         msg = f"Failed value for: {full_name}"
-        # special case if we're comparing list of strings
-        if isinstance(reference, list) and isinstance(reference[0], str):
-            self.assert_equal(actual, reference, name)
-        else:
-            numpy.testing.assert_allclose(actual, reference, rtol=rtol, atol=atol, err_msg=msg)
+        numpy.testing.assert_allclose(actual, reference, rtol=rtol, atol=atol, err_msg=msg)
 
-    def assert_equal(self, actual, reference, name):
```

Comment:
What is the difference between `name` and `full_name` here?

Reply:
`name` is the name of this particular key, while `full_name` includes all the keys in a nested dictionary. Say for example you have
```
{"a": {"b": 1}}
```
`name` would be `b` but `full_name` would be `a: b`.

To be honest I think this stuff is only partially working, we should probably fix it at some point.